src/org/thoughtcrime/securesms/util/SendMessageUtil.java

```diff
@@ -0,0 +1,138 @@
+package org.thoughtcrime.securesms.util;
+
+import android.app.Activity;
+import android.content.Context;
+import android.net.Uri;
+import android.util.Log;
+
+import com.b44t.messenger.DcContext;
+import com.b44t.messenger.DcMsg;
+
+import org.thoughtcrime.securesms.connect.ApplicationDcContext;
+import org.thoughtcrime.securesms.connect.DcHelper;
+import org.thoughtcrime.securesms.mms.PartAuthority;
+import org.thoughtcrime.securesms.providers.PersistentBlobProvider;
+
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Set;
+
+import static android.app.Activity.RESULT_OK;
+import static org.thoughtcrime.securesms.util.RelayUtil.getForwardedMessageIDs;
+import static org.thoughtcrime.securesms.util.RelayUtil.getSharedText;
+import static org.thoughtcrime.securesms.util.RelayUtil.getSharedUris;
+import static org.thoughtcrime.securesms.util.RelayUtil.isForwarding;
+import static org.thoughtcrime.securesms.util.RelayUtil.isSharing;
+
+public class SendMessageUtil {
+  private static final String TAG = SendMessageUtil.class.getSimpleName();
+
+  public static void immediatelyRelay(Activity activity, Set<Long> chatIds) {
+    Util.runOnAnyBackgroundThread(() -> {
+      for (long chatId : chatIds) {
+        immediatelyRelayInner(activity, (int) chatId);
```

Comment:
Why does getListAdapter().getBatchSelections() return a Set<Long>? It has to be converted to `int` here, which is not a problem but definitely bad style.